rasa/utils/tensorflow/model_data_utils.py

```diff
@@ -63,8 +68,20 @@ def featurize_training_examples(
                     attribute, featurizers
                 )
         output.append(attribute_to_features)
-
-    return output
+    # get sparse feature sizes
+    sparse_attributes = []
+    for attr, features in output[0].items():
+        if features and features[0].is_sparse():
+            sparse_attributes.append(attr)
+    # we exclude labels at this point
+    if label_attribute in sparse_attributes:
+        sparse_attributes.remove(label_attribute)
+    feature_sizes = {}
+    for attr in sparse_attributes:
+        feature_sizes[attr] = training_examples[0].get_sparse_feature_sizes(

```

Comment:
Again please be careful when directly indexing over an iterable (`training_examples[0]` can break code for empty lists)